test/test_model.py

```diff
@@ -175,6 +175,18 @@ def test_config(self):
             # Confirm that we cannot modify config values.
             self.model.config['foo'] = 'bar'
 
+    def test_pod_spec(self):
+        os.environ['JUJU_UNIT_NAME'] = 'myapp/0'
+        self.addCleanup(os.environ.pop, 'JUJU_UNIT_NAME')
+        model = op.model.Model('myapp/0', ['db0', 'db1', 'db2'], op.model.ModelBackend())
+
+        fake_script(self, 'pod-spec-set', """cat - > $(dirname $0)/pod_spec.json""")
+        fake_script(self, 'pod-spec-get', """cat $(dirname $0)/pod_spec.json || echo {}""")
+
+        self.assertEqual(model.pod_spec, {})
+        model.pod_spec = {'foo': 'bar'}
+        self.assertEqual(model.pod_spec, {'foo': 'bar'})
```

Comment:
Where can I find more details about these pod specs?

Reply:
https://discourse.jujucharms.com/t/writing-a-kubernetes-charm/159
and
https://discourse.jujucharms.com/t/updated-podspec-yaml-new-features/2124

One thing we should watch out for, is that 'pod-spec-set' has grown '--k8s-resources' for kubernetes specific pod attributes. So there are potentially 2 files that you'd want to manage not just a single blob of text.

Reply:
Hrm.  Regarding the `--k8s-resources`, the current helper just [concatenates the two](https://github.com/juju-solutions/layer-caas-base/blob/master/lib/charms/layer/caas_base.py#L19) and passes them both in via stdin.  That seems like it would lead to ambiguous data and parsing errors, so I converted the code to use files and to handle each separately.